compiler/vmdef.nim

```diff
@@ -261,11 +261,34 @@ type
     config*: ConfigRef
     graph*: ModuleGraph
     oldErrorCount*: int
+    profiler*: Profiler
+
+  PStackFrame* = ref TStackFrame
+  TStackFrame* = object
+    prc*: PSym                 # current prc; proc that is evaluated
+    slots*: seq[TFullReg]      # parameters passed to the proc + locals;
+                              # parameters come first
+    next*: PStackFrame         # for stacking
+    comesFrom*: int
+    safePoints*: seq[int]      # used for exception handling
+                              # XXX 'break' should perform cleanup actions
+                              # What does the C backend do for it?
+  ProfileInfo* = object
+    time*: float
+    count*: int
+
+  Profiler* = object
+    tEnter*: float
+    tos*: PStackFrame
+    data*: Table[TLineInfo, ProfileInfo]
 
   TPosition* = distinct int
 
   PEvalContext* = PCtx
 
+proc hash*(i: TLineInfo): Hash =
+  hash([i.line.int, i.col.int, i.fileIndex.int])
```

Comment:
i really meant: `hash (i.line, i.col, i.fileIndex)`, which is the same as:
`hash((i.line, i.col, i.fileIndex))`
(it works, thanks to hash working with tuples; and doesn't require conversions to a common type as in an array; in particular it works with `distinct`)

Reply:
Ah, got me there! :)